src/vaults.lua

```diff
@@ -123,21 +123,26 @@ end
 --- @param cursor string|nil The address to start from
 --- @param limit number Max number of results to return
 --- @param sortOrder string "asc" or "desc" sort direction
---- @return table Array of {address, vault} objects
-function vaults.getPaginatedVaults(cursor, limit, sortOrder)
+--- @param sortBy string|nil "address", "vaultId", "balance", "startTimestamp", "endTimestamp" field to sort by
+--- @return table Array of { address, vaultId, balance, startTimestamp, endTimestamp } objects
+function vaults.getPaginatedVaults(cursor, limit, sortOrder, sortBy)
 	local allVaults = vaults.getVaults()
 	local vaultsArray = {}
 	local cursorField = "address" -- the cursor will be the wallet address
-	for address, vault in pairs(allVaults) do
-		table.insert(vaultsArray, {
-			address = address,
-			vault = vault,
-		})
+
+	for address, vaultsForAddress in pairs(allVaults) do
```

Comment:
Nit: you could use utils.reduce to convey the canonical higher order function that's taking place here.

Reply:
+1 to this rec